core/services/bulletprooftxmanager/eth_confirmer.go

```diff
@@ -1208,3 +1222,31 @@ func findEthTxWithNonce(db *gorm.DB, fromAddress gethCommon.Address, nonce uint)
 	}
 	return &etx, errors.Wrap(err, "findEthTxsWithNonce failed")
 }
+
+func (ec *EthConfirmer) ResumePendingTaskRuns(ctx context.Context, head models.Head) error {
+	sqlxDB := postgres.UnwrapGormDB(ec.db)
+
+	type x struct {
+		ID      uuid.UUID
+		Receipt []byte
+	}
+	var receipts []x
+	if err := sqlxDB.Select(&receipts, `
+	SELECT pipeline_task_runs.id, eth_receipts.receipt FROM pipeline_task_runs
+	INNER JOIN pipeline_runs ON pipeline_runs.id = pipeline_task_runs.pipeline_run_id
+	INNER JOIN eth_txes ON (eth_txes.meta->>'PipelineTaskRunID')::uuid = pipeline_task_runs.id AND eth_txes.state = 'confirmed'
```

Comment:
This does play nice with a nil meta right? i.e. for keepers I think this join statement just wouldn't match

Reply:
Yeah I tested it both with and without `WHERE eth_txes.meta->>'PipelineTaskRunID' IS NOT NULL` and it seemed to work fine without, just like a regular INNER JOIN where a field is null.